app/models/fe/element.rb

```diff
@@ -119,7 +119,7 @@ def previous_element(question_sheet, page = nil)
     end
 
     def hidden_by_conditional?(answer_sheet, page)
-      prev_el = previous_element(answer_sheet.question_sheet, page)
+      prev_el = previous_element(page.question_sheet, page) if answer_sheet.question_sheets.include?(page.question_sheet)
```

Comment:
Or can we do

```
return false unless answer_sheet.question_sheets.include?(page.question_sheet)
```

as a sort of sanity check?  an element will never be hidden if it's not even on the page passed in